### PR TITLE
Fix Typos in related-files-fn doc

### PR DESCRIPTION
**Pull request by LaurenceWarne**
Hi! This PR just fixes some minor typos in the doc for `related-files-fn`.

-----------------

Before submitting a PR make sure the following things have been done (and denote this
by checking the relevant checkboxes):

- [x] The commits are consistent with our [contribution guidelines](../blob/master/CONTRIBUTING.md)
- [ ] You've added tests (if possible) to cover your change(s)
- [x] All tests are passing ([`eldev test`](https://github.com/doublep/eldev))
- [x] The new code is not generating bytecode or `M-x checkdoc` warnings
- [ ] You've updated the [changelog](../blob/master/CHANGELOG.md) (if adding/changing user-visible functionality)
- [ ] You've updated the readme (if adding/changing user-visible functionality)

Thanks!

### CODE REVIEW

File: doc/modules/ROOT/pages/projects.adoc [hunk 1]
```diff
@@ -431,7 +431,7 @@ Each helper means `projectile-related-files-fn-helper-name` function.
 
 (projectile-register-project-type
    ;; ...
-   :related-files-fn #'my/related-files)
```
Comment (bbatsov) [inline doc/modules/ROOT/pages/projects.adoc:434]: This definitely doesn't seem like a typo to me.

Comment (LaurenceWarne) [reply doc/modules/ROOT/pages/projects.adoc:434]: `my/related-files` is a list here, so I find using `:related-files-fn #'my/related-files` results in an `unsupported value type of related-files-fn`, or am I missing something here?

Comment (bbatsov) [reply doc/modules/ROOT/pages/projects.adoc:434]: That's supposed to be a function, not a list.

Comment (LaurenceWarne) [reply doc/modules/ROOT/pages/projects.adoc:434]: But `related-files'fn` can accept a list as well:
https://github.com/bbatsov/projectile/blob/793abfead25a7ef92ad3a443ad3067a82f4a2fb9/projectile.el#L2417
 which is what this example is showing?

Comment (bbatsov) [reply doc/modules/ROOT/pages/projects.adoc:434]: Oh, my bad. I should have expanded the diff.

**Comment by bbatsov** [comment]
Thanks!

**Pull request merged**